evaluator/src/main/kotlin/SourceTreeResolver.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2022 EPAM Systems, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.evaluator
+
+import java.io.File
+
+import org.apache.logging.log4j.kotlin.Logging
+
+import org.ossreviewtoolkit.downloader.Downloader
+import org.ossreviewtoolkit.model.Package
+import org.ossreviewtoolkit.model.VcsInfo
+import org.ossreviewtoolkit.model.config.DownloaderConfiguration
+import org.ossreviewtoolkit.utils.ort.createOrtTempDir
+
+fun interface SourceTreeResolver {
+    companion object : Logging {
+        fun forRemoteRepository(vcsInfo: VcsInfo) =
+            SourceTreeResolver {
+                logger.info {
+                    "Downloading ${vcsInfo.type} source code repository from ${vcsInfo.url}..."
+                }
+
+                val downloadDir = createOrtTempDir()
+                val downloader = Downloader(DownloaderConfiguration())
```

Comment:
You should set `DownloaderConfiguration.sourceCodeOrigin = listOf(SourceCodeOrigin.VCS)` to be explicit, otherwise if the download fails there will be additional logs about the failed source artifact download which could be confusing as projects never have a source artifact.